packages/libs/src/services/discoveryProvider/DiscoveryProvider.ts

```diff
@@ -820,8 +839,21 @@ export class DiscoveryProvider {
    * Return user collections (saved & uploaded) along w/ users for those collections
    */
   async getUserAccount(wallet: string) {
-    const req = Requests.getUserAccount(wallet)
-    return await this._makeRequest<CurrentUser>(req)
+    const userIdOverride = this.enableUserIdOverride
+      ? await getUserIdOverride(this.localStorage)
+      : undefined
+    // If override is used, fetch that account instead
+    if (userIdOverride) {
+      const req = Requests.getUsers(1, 0, [parseInt(userIdOverride)])
+      const res = await this._makeRequest<CurrentUser[]>(req)
+      if (res && res.length > 0 && res[0]) {
+        return { ...res[0], playlists: [] } as CurrentUser
```

Comment:
why do we have to do this? (might be good to put a comment in the code)
I'm guessing `getUserAccount` adds the `playlists` thing to the object while the plain `getUsers` doesn't? Will we need to populate `playlists` ourselves in the future then?

Reply:
Correct. The user metadata object shared between the two defines the playlists as optional because fetching the "account" gets you a decorated user object that includes playlists but fetching any other user record through the standard endpoint does not include that info.

This is part of the reason this is a temporary hack, because it won't return the same state we currently use. I think the longer term solution here is to split the fetching of account playlists to a separate API call and remove them from the `Usermetadata` object. Or we could do a parallel fetch to two endpoints at this level and merge the results into the expected shape.